joplin/events/models.py

```diff
@@ -0,0 +1,157 @@
+from django.db import models
+from wagtail.core.blocks import StructBlock, PageChooserBlock, TextBlock
+from modelcluster.fields import ParentalKey
+from wagtail.admin.edit_handlers import (
+    InlinePanel,
+    FieldRowPanel,
+    StreamFieldPanel,
+    MultiFieldPanel,
+)
+
+from locations.models import LocationPage
+from base.models import Contact
+from wagtail.core.fields import RichTextField, StreamField
+from wagtail.core.models import Orderable
+from wagtail.admin.edit_handlers import FieldPanel, PageChooserPanel, HelpPanel
+from base.models import JanisBasePage
+from base.models.widgets import countMe, countMeLongTextArea, AUTHOR_LIMITS
+from modelcluster.models import ClusterableModel
+from base.models.constants import DEFAULT_MAX_LENGTH, WYSIWYG_GENERAL
+from wagtail.snippets.edit_handlers import SnippetChooserPanel
+
+
+class EventPage(JanisBasePage):
+    janis_url_page_type = 'event'
+
+    description = RichTextField(
+        features=WYSIWYG_GENERAL,
+        verbose_name='Long description',
+        help_text='Full description of the event',
+        blank=True
+    )
+
+    date = models.DateField(verbose_name="Event date", blank=True, null=True)
+    start_time = models.TimeField(blank=True, null=True)
+    end_time = models.TimeField(blank=True, null=True)
+
+    location_blocks = StreamField(
+        [
+            ('city_location', StructBlock(
+                [
+                    ('location_page', PageChooserBlock(label="Location", page_type=[LocationPage], classname='do-not-hide')),
+                    ('additional_details_en', TextBlock(label='Any other necessary location details, such as room number [en]', required=False)),
+                    ('additional_details_es', TextBlock(label='Any other necessary location details, such as room number [es]', required=False)),
+                    ('additional_details_ar', TextBlock(label='Any other necessary location details, such as room number [ar]', required=False)),
+                    ('additional_details_vi', TextBlock(label='Any other necessary location details, such as room number [vi]', required=False)),
+                ]
+            )),
+            ('remote_location', StructBlock(
```

Comment:
I realize this is a minor quibble, but I think the name 'remote location' could be kinda confusing. It makes me think that the event could be a webinar or somehow not in person. 

I know content side its described as 'non-city location', so I'd expect it to be called something like 'non_city_location', 'NonCityLocation', etc. It'l make it much easier to find in the codebase in the future.

Reply:
@desigonz and I discussed and the plan is to have content come up with a final name, but to use remote location until we have a decision from them. I'll add some comments to the code.